jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -682,7 +683,64 @@ ecma_typedarray_create_object_with_typedarray (ecma_object_t *typedarray_p, /**<
 } /* ecma_typedarray_create_object_with_typedarray */
 
 /**
- * Create a TypedArray object by transforming from an array-like object
+ * Helper method for ecma_op_typedarray_from
+ *
+ * @return ECMA_VALUE_TRUE - if setting the given value to the new typedarray was successful
+ *         ECMA_VALUE_ERROR - otherwise
+ */
+static ecma_value_t
+ecma_op_typedarray_from_helper (ecma_value_t this_val, /**< this_arg for the above from function */
+                                ecma_value_t current_value, /**< given value to set */
+                                uint32_t index, /**< currrent index */
+                                ecma_object_t *func_object_p, /**< map function object */
+                                ecma_typedarray_info_t *info_p, /**< typedarray info */
+                                ecma_typedarray_setter_fn_t setter_cb) /**< setter callback function */
+{
+  ecma_value_t mapped_value;
+  if (func_object_p != NULL)
```

Comment:
AFAIK, this case is not tested. There should be a test which uses the "mapfn" from the standard.